pkg/apis/build/v1alpha2/cluster_lifecycle_types.go

```diff
@@ -0,0 +1,81 @@
+package v1alpha2
+
+import (
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+
+	corev1alpha1 "github.com/pivotal/kpack/pkg/apis/core/v1alpha1"
+)
+
+const (
+	ClusterLifecycleKind   = "ClusterLifecycle"
+	ClusterLifecycleCRName = "clusterlifecycles.kpack.io"
+)
+
+// +genclient
+// +genclient:nonNamespaced
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object,k8s.io/apimachinery/pkg/apis/meta/v1.ObjectMetaAccessor
+
+// +k8s:openapi-gen=true
+type ClusterLifecycle struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	Spec   ClusterLifecycleSpec   `json:"spec"`
+	Status ClusterLifecycleStatus `json:"status"`
+}
+
+// +k8s:openapi-gen=true
+type ClusterLifecycleSpec struct {
+	// +listType
+	corev1alpha1.ImageSource `json:",inline"`
+	ServiceAccountRef        *corev1.ObjectReference `json:"serviceAccountRef,omitempty"`
+}
+
+// +k8s:openapi-gen=true
+type ClusterLifecycleStatus struct {
+	corev1alpha1.Status      `json:",inline"`
+	ResolvedClusterLifecycle `json:",inline"`
+}
+
+// +k8s:openapi-gen=true
+type ResolvedClusterLifecycle struct {
+	Version string `json:"version,omitempty"`
+
+	// Deprecated: Use `LifecycleAPIs` instead
+	API  LifecycleAPI  `json:"api,omitempty"`
```

Comment:
Do we need this if it's deprecated?

Reply:
Maybe not. I was trying to change as little as possible, but I can rip it out if you are okay with that.

Reply:
Updated in 77c63219

Reply:
Discussed offline, to minimize the surface area of changes we will leave this in the builder metadata label